events_task/lib/my_app/accounts/accounts.ex

```diff
@@ -12,8 +8,15 @@ defmodule MyApp.Accounts do
     Repo.all(User)
   end
 
-  def get_user(id) do
-    Repo.get(User, id)
+  def get_user(param) when not is_map(param) do
+    Repo.get(User, param)
+  end
+
+  def get_user(param) when is_map(param) do
+    case Enum.at(Map.keys(param), 0) do
```

Comment:
So if somebody called `get_user(email: "test@example.com", id: 5)`, you will ignore second param and only look by email? This doesn't look like an expected thing :).

It seems that you wanted three different cases to get user by. You can implement it like this:

```
def get_user(id) when is_integer(id), do: Repo.get(User, id)
def get_user(:id, id) when is_integer(id), do: Repo.get(User, id)
def get_user(:email, email) when is_binary(email), do: Repo.get_by(User, email: email)
```

Second function case is not required.

Or, you could give up trying to wrap getting a user, and just pass things to Repo.get_by:
```
def get_user(id) when is_integer(id), do: Repo.get(Repo, id)
def get_user(params) when is_map(params), do: Repo.get_by(Repo, params)
```

Both options would be fine.